internal/worker/sshserver/server_test.go

```diff
@@ -180,13 +184,93 @@ func (s *sshServerSuite) TestSSHServer(c *gc.C) {
 
 	output, err := terminatingSession.CombinedOutput("")
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(string(output), gc.Equals, "Your final destination is: 1.postgresql.8419cd78-4993-4c3a-928e-c646226beeee.juju.local as user: ubuntu\n")
+	c.Assert(string(output), gc.Equals, fmt.Sprintf("Your final destination is: 8419cd78-4993-4c3a-928e-c646226beeee as user: ubuntu\n"))
 
 	// Server isn't gracefully closed, it's forcefully closed. All connections ended
 	// from server side.
 	workertest.CleanKill(c, server)
 }
 
+func (s *sshServerSuite) TestSSHPublicKeyHandler(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	listener := bufconn.Listen(8 * 1024)
+
+	s.facadeClient.EXPECT().VerifyPublicKey(gomock.Any()).DoAndReturn(func(sshPKIAuthArgs params.VerifyPublicKeyArgs) error {
+		if strings.Contains(sshPKIAuthArgs.ModelUUID, "8419cd78-4993-4c3a-928e-c646226beeee") {
+			return nil
+		}
+		return errors.NotFound
+	}).AnyTimes()
+	s.facadeClient.EXPECT().HostKeyForTarget(gomock.Any()).Return(s.hostKey, nil).AnyTimes()
+
+	server, err := sshserver.NewServerWorker(sshserver.ServerWorkerConfig{
+		Logger:                   loggo.GetLogger("test"),
+		Listener:                 listener,
+		JumpHostKey:              jujutesting.SSHServerHostKey,
+		FacadeClient:             s.facadeClient,
+		NewSSHServerListener:     newTestingSSHServerListener,
+		MaxConcurrentConnections: maxConcurrentConnections,
+	})
+	c.Assert(err, gc.IsNil)
+	defer workertest.DirtyKill(c, server)
+
+	userKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	c.Assert(err, gc.IsNil)
+
+	notValidSigner, err := ssh.NewSignerFromKey(userKey)
+	c.Assert(err, gc.IsNil)
+
+	tests := []struct {
+		name               string
+		destinationAddress string
+		key                ssh.Signer
+		expectSuccess      bool
+	}{
+		{
+			name:               "valid destination model uuid and public key",
+			destinationAddress: testVirtualHostname,
+			key:                s.userSigner,
+			expectSuccess:      true,
+		},
+		{
+			name:               "model uuid not valid",
+			destinationAddress: "1.postgresql.8419cd78-4993-4c3a-928e-eeeeeeeeeeee.juju.local",
+			key:                notValidSigner,
```

Comment:
It wouldn't change anything to use a valid key here would it? It's already an invalid model so you can reduce some lines from this test by using s.userSigner.